src/openforms/analytics_tools/utils.py

```diff
@@ -130,11 +125,13 @@ def update_analytical_cookies(
         Cookie.objects.filter(name__in=[cookie["name"] for cookie in cookies]).delete()
 
 
-def update_csp(csps: list[CSPDict], create: bool):
+def update_csp(
+    config_model: "AnalyticsToolsConfiguration",
+    csps: list[CSPEntry],
+    identifier: "AnalyticsTools",
+    create: bool,
+):
     if create:
-        instances = [
-            CSPSetting(directive=csp["directive"], value=csp["value"]) for csp in csps
-        ]
-        CSPSetting.objects.bulk_create(instances)
+        CSPSetting.objects.set_for(config_model, settings=csps, identifier=identifier)
     else:
-        CSPSetting.objects.filter(value__in=[csp["value"] for csp in csps]).delete()
+        CSPSetting.objects.delete_for(config_model, identifier=identifier)
```

Comment:
I'd say that the caller needs to properly pass teh `csps` list:

```py
csps = [] if not create else [...]
update_csp(config, csps, identifier)
```
and at this point, this `update_csp` util should probably be skipped entirely since it really just wraps `CSPSetting.objects.set_for(...)`

Setting an empty list of settings is equivalent to `CSPSetting.objects.delete_for`, which means that `delete_for` should not exist.

Reply:
Simplifies things a lot indeed.